src/server/mod.rs

```diff
@@ -1508,7 +1507,7 @@ impl LanguageServer for LspServer {
                             if response.applied {
                                 self.store.put(
                                     &command_params.text_document.uri,
-                                    &new_text,
+                                    composition.to_string().as_str(),
```

Comment:
Pretty sure we are supposed to be putting in the store the entire editor text, and not just the replacement composition block text. So I updated this everywhere.

Reply:
Oh, huh. I didn't even know this block of code existed. Do we not also get a `textDocument/onChange` event?

Reply:
Yes, the LSP sends that next. Therefore, we could remove this update to the store...assuming we want the monaco-editor to be the src for all of those updates.